thetis/callback.py

```diff
@@ -684,10 +706,31 @@ def _initialize(self):
         xyz = (self.x, self.y, self.z) if self.on_sphere else (self.x, self.y)
         self.xyz = numpy.array([xyz])
 
+        # create VertexOnlyMesh and function spaces
+        self.mesh = self.solver_obj.fields[self.fieldnames[0]].function_space().mesh()
+        self.vom = VertexOnlyMesh(self.mesh, self.xyz, redundant=True)
+        self.function_spaces = {}
+        for field_name in self.fieldnames:
+            field = self.solver_obj.fields[field_name]
+            if isinstance(field.function_space().ufl_element(), VectorElement):
+                P0DG = VectorFunctionSpace(self.vom, "DG", 0)
+                P0DG_input_ordering = VectorFunctionSpace(self.vom.input_ordering, "DG", 0)
+            else:
+                P0DG = FunctionSpace(self.vom, "DG", 0)
+                P0DG_input_ordering = FunctionSpace(self.vom.input_ordering, "DG", 0)
+            self.function_spaces[field_name] = (P0DG, P0DG_input_ordering)
+
         # test evaluation
         try:
             if self.eval_func is None:
-                self.solver_obj.fields.bathymetry_2d.at(self.xyz, tolerance=self.tolerance)
+                bathymetry_field = self.solver_obj.fields.bathymetry_2d
+                P0DG = FunctionSpace(self.vom, "DG", 0)
+                P0DG_input_ordering = FunctionSpace(self.vom.input_ordering, "DG", 0)
+                f_at_points = Function(P0DG)
+                f_at_input_points = Function(P0DG_input_ordering)
+                f_at_points.interpolate(bathymetry_field)
+                f_at_input_points.interpolate(f_at_points)
```

Comment:
fyi iirc this is just to check that the interpolation works (points aren't outside the domain) and is included in initialization to fail early rather than waiting for the first usage (which typically only happens after some iterations). Any field could be interpolated, and the result can be discarded

Reply:
Ah, that makes sense - thanks